### PR TITLE
CHS OAuth2 clients

### PR DESCRIPTION
**Pull request by matteoformica-amido**
# Description

added OAuth2 clients for API Filing M2M and CHS webapps, renamed API Filing WEB

**FIDC Update Required:**
- [x] applications
- [ ] am-scripts
- [ ] auth-trees
- [ ] connectors
- [ ] cors
- [ ] idm-access-config
- [ ] idm-endpoints
- [ ] managed-objects
- [ ] password-policy
- [ ] services
- [ ] internal-roles

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ that will require FR to update FIDC
 environment specific variables.
 -->
 **FIDC Update Required:**
+- [ ] applications
```
Comment (saulfrance) [inline .github/pull_request_template.md:11]: Don't think we need this. Application updates are applied by the pipeline

File: config/applications/api-filing-m2m-client.json.tpl [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+			"inherited": false,
+			"value": 604800
+		},
+		"scopes": {
+			"inherited": false,
+			"value": [
+                "openid",
+				"profile",
+				"email",
+                "https://account.companieshouse.gov.uk/user/profile.read",
```
Comment (steveclewer-amido) [inline config/applications/api-filing-m2m-client.json.tpl:23]: Format as per others, if I'm being picky

**Review by steveclewer-amido — APPROVED** [review]

File: config/applications/api-filing-m2m-client.json.tpl [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+				"https://identity.company-information.service.gov.uk/user/profile.read",
+				"https://api.companieshouse.gov.uk/company/registered-office-address.update",
+				"https://api.company-information.service.gov.uk/company/registered-office-address.update",
+				"https://api.companieshouse.gov.uk/company/admin.write-full",
+				"https://api.company-information.service.gov.uk/company/admin.write-full"
```
Comment (saulfrance) [inline config/applications/api-filing-m2m-client.json.tpl:28]: Will these need to be different in each in environment?

File: config/applications/chs-internal-client.json.tpl [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+		},
+		"customProperties": {
+			"inherited": false,
+			"value": []
+		},
+		"javascriptOrigins": {
+			"inherited": false,
+			"value": [
+                "https://idam.amido.aws.chdev.org", 
+                "https://idam.amido.aws.chdev.org:443"
```
Comment (saulfrance) [inline config/applications/chs-internal-client.json.tpl:113]: You can use the existing `UI_URL` variable for these

File: config/applications/api-filing-m2m-client.json.tpl (hunk 3 above)
Comment (matteoformica-amido) [reply config/applications/api-filing-m2m-client.json.tpl:28]: I have no idea...

**Pull request merged**